detector/src/main/java/com/synopsys/integration/detector/result/DetectorResult.java

```diff
@@ -45,8 +51,7 @@ public String getDescription() {
         return description;
     }
 
-    public String getResultClassName() {
-        return resultClassName;
+    public Class getResultClass() {
+        return resultClass;
```

Comment:
In the case resultClass is null, we could return this.getClass();

Reply:
I like that